### ISSUE TITLE
[Bug]: [Nightly]Milvus crashed reporting 'fatal error: unexpected signal during runtime execution'

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#21001]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:039e9ce
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus2.3.0.dev15
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Milvus crashed reporting 'fatal error: unexpected signal during runtime execution'.
```
2022-12-06T02:55:49.703587255+08:00 stdout F 2022-12-05 18:55:49,703 | WARNING | default | io_getevents() failed; returned 2, expected=8, ernno=-2=Unknown error -2, try #2
2022-12-06T02:55:49.712525148+08:00 stderr F fatal error: unexpected signal during runtime execution
2022-12-06T02:55:49.714680815+08:00 stderr F [signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x7f3b9f6a28c1]
2022-12-06T02:55:49.714720798+08:00 stderr F
2022-12-06T02:55:49.714732455+08:00 stderr F runtime stack:
2022-12-06T02:55:49.714742579+08:00 stderr F runtime.throw({0x349bdbd?, 0xb100000204?})
2022-12-06T02:55:49.714751059+08:00 stderr F    /usr/local/go/src/runtime/panic.go:992 +0x71
2022-12-06T02:55:49.714760659+08:00 stderr F runtime.sigpanic()
2022-12-06T02:55:49.714773352+08:00 stderr F    /usr/local/go/src/runtime/signal_unix.go:802 +0x3a9
2022-12-06T02:55:49.714781352+08:00 stderr F
2022-12-06T02:55:49.714790799+08:00 stderr F goroutine 18223 [syscall]:
2022-12-06T02:55:49.714819039+08:00 stderr F runtime.cgocall(0x2c6f0c0, 0xc001264c38)
2022-12-06T02:55:49.714838788+08:00 stderr F    /usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc001264c10 sp=0xc001264bd8 pc=0x116f7bc
2022-12-06T02:55:49.714847209+08:00 stderr F github.com/milvus-io/milvus/internal/querynode._Cfunc_AppendIndex(0x7f3354014b60, 0x7f3354011c40)
2022-12-06T02:55:49.714854691+08:00 stderr F    _cgo_gotypes.go:178 +0x55 fp=0xc001264c38 sp=0xc001264c10 pc=0x2ad3755
2022-12-06T02:55:49.714861081+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendIndexData.func5(0x7f3354010e80?, 0x3449e67?)
2022-12-06T02:55:49.71487969+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/querynode/load_index_info.go:199 +0x65 fp=0xc001264c78 sp=0xc001264c38 pc=0x2add445
2022-12-06T02:55:49.714894737+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendIndexData(0xc001264e00?, {0x0, 0x0, 0xc0034948b8?}, {0xc00077a180, 0x7, 0x7f3354014d50?})
2022-12-06T02:55:49.714904238+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/load_index_info.go:199 +0x145 fp=0xc001264d20 sp=0xc001264c78 pc=0x2add165
2022-12-06T02:55:49.714922772+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendLoadIndexInfo(0x0?, {0x0, 0x0, 0x0}, 0xc00262c280, 0x22?, 0x0?, 0xa?, 0x203000?)
```

### Expected Behavior

succeed

### Steps To Reproduce

```markdown
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/211/pipeline/143/
```

### Milvus Log

[artifacts-milvus-distributed-kafka-nightly-211-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/211/artifact/artifacts-milvus-distributed-kafka-nightly-211-pymilvus-e2e-logs.tar.gz)
log:
1. ci.var.log.containers.mdk-211-n-milvus-querynode-78f5f44b75-2pvst_milvus-ci_querynode-e02b3038bf677c4481bd4156f619026605053f56dbd38242a8e42d9b63bdc60b.log
2. ci.var.log.containers.mdk-211-n-milvus-querynode-78f5f44b75-2pvst_milvus-ci_querynode-e02b3038bf677c4481bd4156f619026605053f56dbd38242a8e42d9b63bdc60b.log

**Comment by xiaofan-luan** [comment]
seems to be the same diskANN issue. 
@yanliang567 we need to reproduce it with knowhere only

**Comment by yanliang567** [comment]
/assign @elstic  @liliu-z 
could you please help to reproduce it in knowhere?

/unassign

**Comment by NicoYuan1986** [comment]
Fixed in master. I'll close the issue with it fixed in 2.2 branch.

**Comment by liliu-z** [comment]
> Fixed in master. I'll close the issue with it fixed in 2.2 branch.

Cherry-picked

**Comment by NicoYuan1986** [comment]
2.2 fixed in #21241
